src/main/scala/org/hammerlab/guacamole/distributed/RegionsByContig.scala

```diff
@@ -20,22 +20,30 @@ import org.hammerlab.guacamole.reference.ReferenceRegion
  * @param regionIterator regions, sorted by contig and start locus.
  */
 class RegionsByContig[R <: ReferenceRegion](regionIterator: Iterator[R]) {
+
   private val buffered = regionIterator.buffered
   private var seenContigs = List.empty[String]
-  private var prevIterator: Option[SingleContigRegionIterator[R]] = None
+  private var curIterator: SingleContigRegionIterator[R] = _
+
   def next(contig: String): Iterator[R] = {
-    // We must first march the previous iterator we returned to the end.
-    while (prevIterator.exists(_.hasNext)) prevIterator.get.next()
```

Comment:
Seems like potentially more than a stylistic change. Is this intended to change semantics?